src/server/req_modify.c

```diff
@@ -844,13 +850,15 @@ req_modifyReservation(struct batch_request *preq)
 						req_reject(PBSE_BADTSPEC, 0, preq);
 					return;
 				}
+				
+				if (d==1) {
+					presv->ri_alter_etime = presv->ri_wattr[RESV_ATR_end].at_val.at_long;
+					presv->ri_wattr[RESV_ATR_end].at_flags &= ~ATR_VFLAG_SET;
```

Comment:
Why are you unsetting the end of the reservation?

Reply:
start_end_dur_wall() function calculates the value of field with unset flag for us, so the need to unset the flag

Reply:
But if this function fails, the end of the reservation is now unset forever.

Reply:
For this I have updated the resv_revert_alter_times() function. In case the function fails resv_revert_alter_times() will reset the flags and also the starttime and endtime to the previous value